File: tournament/tournament/Infrastructure/DataBase/Models/Match.cs
using System;
using System.Collections.Generic;

namespace tournament.Infrastructure.DataBase.Models
{
    public class Match : BaseEntity
    {
        public int SequenceNr { get; set; }
        public int WinnerId { get; set; }
        public DateTime Date { get; set; }
        public int ScoreTeam1 { get; set; }
        public int ScoreTeam2 { get; set; }
        public ICollection<MatchTeam> Teams { get; set; }
        public Tournament Tournament { get; set; }
        // Foreign Keys
        public int TournamentId { get; set; }
        public int? WinnerGoesToId { get; set; }
        public int? LoserGoesToId { get; set; }
    }

}

File: tournament-ui/src/app/models/match.ts
import { Team } from "./team";

export class Match {
    id: number;
    tournamentId: number;
    sequenceNr : number;
    //team1: Team;
    //team2: Team;
    winnerId: number;
    loserId: number;
    winnerGoesToId: number;
    loserGoesToId: number;
    scoreTeam1: number;
    scoreTeam2: number;
    matchTeams: Team[];
}
